cmd/beekeeper/cmd/restart.go

```diff
@@ -12,60 +12,65 @@ import (
 
 func (c *command) initRestartCmd() (err error) {
 	const (
-		optionNameLabelSelector = "label-selector"
-		optionNameNamespace     = "namespace"
-		optionNameImage         = "image"
-		optionNameNodeGroups    = "node-groups"
-		optionNameTimeout       = "timeout"
+		optionNameLabelSelector  = "label-selector"
+		optionNameNamespace      = "namespace"
+		optionNameImage          = "image"
+		optionNameNodeGroups     = "node-groups"
+		optionNameTimeout        = "timeout"
+		optionNameDeploymentType = "deployment-type"
 	)
 
 	cmd := &cobra.Command{
 		Use:   "restart",
 		Short: "Restart pods in a cluster or namespace",
 		Long:  `Restarts pods by deleting them. Uses cluster name as the primary scope or falls back to namespace, with optional label filtering.`,
 		RunE: func(cmd *cobra.Command, args []string) (err error) {
-			ctx, cancel := context.WithTimeout(cmd.Context(), c.globalConfig.GetDuration(optionNameTimeout))
-			defer cancel()
+			return c.withTimeoutHandler(cmd, func(ctx context.Context) error {
+				clusterName := c.globalConfig.GetString(optionNameClusterName)
+				namespace := c.globalConfig.GetString(optionNameNamespace)
 
-			clusterName := c.globalConfig.GetString(optionNameClusterName)
-			namespace := c.globalConfig.GetString(optionNameNamespace)
-
-			if clusterName == "" && namespace == "" {
-				return errors.New("either cluster name or namespace must be provided")
-			}
-
-			restartClient := restart.NewClient(c.k8sClient, c.log)
-
-			if clusterName != "" {
-				clusterConfig, ok := c.config.Clusters[clusterName]
-				if !ok {
-					return fmt.Errorf("cluster config %s not defined", clusterName)
+				if clusterName == "" && namespace == "" {
+					return errors.New("either cluster name or namespace must be provided")
 				}
 
-				cluster, err := c.setupCluster(ctx, clusterName, false)
+				nodeClient, err := c.createNodeClient(ctx, true)
 				if err != nil {
-					return fmt.Errorf("setting up cluster %s: %w", clusterName, err)
+					return fmt.Errorf("creating node client: %w", err)
 				}
 
-				c.log.Infof("restarting cluster %s", clusterName)
+				restartClient := restart.NewClient(nodeClient, c.k8sClient, c.log)
 
-				if err := restartClient.RestartCluster(ctx,
-					cluster,
-					clusterConfig.GetNamespace(),
-					c.globalConfig.GetString(optionNameImage),
-					c.globalConfig.GetStringSlice(optionNameNodeGroups),
-				); err != nil {
-					return fmt.Errorf("restarting cluster %s: %w", clusterName, err)
-				}
+				if clusterName != "" {
+					clusterConfig, ok := c.config.Clusters[clusterName]
+					if !ok {
+						return fmt.Errorf("cluster config %s not defined", clusterName)
+					}
 
-				return nil
-			}
+					cluster, err := c.setupCluster(ctx, clusterName, false)
```

Comment:
`c.setupCluster(ctx, clusterName, false)` is already called in `c.createNodeClient(ctx, true)`, no need to call it twice.